src/planner/vectorfield/MoveHandStraightVectorField.cpp

```diff
@@ -0,0 +1,142 @@
+#include "MoveHandStraightVectorField.hpp"
+#include <fstream>
+#include <iostream>
+#include <boost/format.hpp>
+#include <dart/dynamics/BodyNode.hpp>
+#include <dart/math/MathTypes.hpp>
+#include <dart/optimizer/Function.hpp>
+#include <dart/optimizer/Problem.hpp>
+#include <aikido/planner/vectorfield/VectorFieldUtil.hpp>
+
+namespace aikido {
+namespace planner {
+namespace vectorfield {
+
+MoveHandStraightVectorField::MoveHandStraightVectorField(
+    dart::dynamics::BodyNodePtr bn,
+    Eigen::Vector3d const& linearVelocity,
+    double minDuration,
+    double maxDuration,
+    double stepsize,
+    double linearGain,
+    double linearTolerance,
+    double rotationGain,
+    double rotationTolerance,
+    double optimizationTolerance,
+    double padding)
+  : bodynode_(bn)
+  , velocity_(linearVelocity)
+  , linearDirection_(linearVelocity.normalized())
+  , minDuration_(minDuration)
+  , maxDuration_(maxDuration)
+  , timestep_(stepsize)
+  , linearGain_(linearGain)
+  , linearTolerance_(linearTolerance)
+  , rotationGain_(rotationGain)
+  , rotationTolerance_(rotationTolerance)
+  , optimizationTolerance_(optimizationTolerance)
+  , padding_(padding)
+  , startPose_(bn->getTransform())
+{
+  assert(minDuration_ >= 0);
+  assert(maxDuration_ >= minDuration_);
+  assert(timestep_ >= 0);
+  assert(velocity_.all() >= 0);
+  assert(linearGain_ >= 0);
+  assert(linearTolerance_ > 0);
+  assert(rotationGain_ >= 0);
+  assert(rotationTolerance_ > 0);
+  assert(optimizationTolerance_ > 0);
+
+  targetPose_ = startPose_;
+  targetPose_.translation() += velocity_ * maxDuration_;
+  maxDuration_ += padding;
+}
+
+bool MoveHandStraightVectorField::operator()(
```

Comment:
Nit: Please use the splitter between function definitions in source files:
`//==============================================================================`